curvefs/src/mds/topology/topology_manager.cpp

```diff
@@ -1064,6 +1064,94 @@ void TopologyManager::GetMetaServersSpace(
     topology_->GetMetaServersSpace(spaces);
 }
 
+void TopologyManager::GetTopology(ListTopologyResponse* response) {
+    // cluster info
+    ClusterInformation info;
+    if (topology_->GetClusterInfo(&info)) {
+        response->set_clusterid(info.clusterId);
+    } else {
+        response->set_clusterid("unknown");
+    }
+
+    // pool
+    ListPool(nullptr, response->mutable_pools());
+    // zone
+    ListZone(response->mutable_zones());
+    // Server
+    ListServer(response->mutable_servers());
+    // MetaServer
+    ListMetaserverOfCluster(response->mutable_metaservers());
+}
+
+void TopologyManager::ListZone(ListZoneResponse* response) {
+    response->set_statuscode(TopoStatusCode::TOPO_OK);
+    auto zoneIdVec = topology_->GetZoneInCluster();
+    for (auto const& zoneId : zoneIdVec) {
+        Zone zone;
+        if (topology_->GetZone(zoneId, &zone)) {
+            auto zoneInfo = response->add_zoneinfos();
+            zoneInfo->set_zoneid(zone.GetId());
+            zoneInfo->set_zonename(zone.GetName());
+            zoneInfo->set_poolid(zone.GetPoolId());
+        } else {
+            LOG(ERROR) << "Topology has counter an internal error: "
+                       << "[func:] ListZone, "
+                       << "[msg:] Zone not found, id = " << zoneId;
+            response->set_statuscode(TopoStatusCode::TOPO_INTERNAL_ERROR);
```

Comment:
you didn't set response ok in the if branch

Reply:
has set in start